### ISSUE TITLE
Request a Feature

### ISSUE THREAD

**Issue by glpaper06** [tetsuwo/website-blocker-chrome.ext#49]
Would it be possible to have only certain areas of a website be blocked?  For example, if I want all of reddit blocked I would normally put "reddit.com 0930-2300" into the input box.  

Is it possible to have a section that overrides it ONLY if a certain combination of words is found in the URL?  So I would put "reddit.com 0930-2300"  into the input box, but in the override box I would put "http://www.reddit.com/r/leagueoflegends/", which is a separate section of the overall website, and anything with "http://www.reddit.com/r/leagueoflegends/" in its URL would be exempt from being blocked?  

I think it would be a nice feature, seeing as on some websites I want the majority of it to not be a distraction, but some of it is nice to be able to use.

**Comment by Libervurto** [comment]
I agree that this would be a useful feature, however, I am a fan of the scripting style of the extension and would prefer to see that expanded upon rather than introduce special lists.  I think even the new 'days of the week' feature should have been implemented as part of the scripting.  e.g.

`twitter.com 1000-1259 mon`

would block only twitter for a time period on Mondays.  Alternatively one could block twitter for the entirety of Monday: 

`twitter.com mon`

Or we could block twitter for the working week:

`twitter.com 0900-1700 mon-fri`

White-listing could also be done via scripting, by prefixing a domain or partial URL with '!' (commonly used to mean 'not').  Using the reddit example above we could achieve the desired block with something like this

`reddit.com !/r/leagueoflegends 0930-2300`

So this line means: if the URL contains reddit.com block it unless it also contains /r/leagueoflegends.

we can use the '!' to great effect with other parameters too.  Say we only want to allow facebook visits between 1700 and 2200, normally this would require blocking the rest of the day like so

`facebook.com 0000-1659,2200-2359`

which is a bit messy and not immediately obvious, but utilising '!' the same thing would look like this

`facebook.com !1700-2200`

One example I would find '!' useful is that after certain time at night I only want .org and .ac.uk sites to be accessible.  I don't think this is possible with the extension in its current state but it could be achieved like so:

`!(.org|.ac.uk) !0800-2300`

See I even used another '!' for the time because it was easier to work out that way.
